users/views.py

```diff
@@ -1,3 +1,71 @@
-from django.shortcuts import render
+import json
+import re
+import bcrypt
+import jwt
 
-# Create your views here.
+from django.http      import JsonResponse, HttpResponse
+from django.views     import View
+from users.models     import User
+from my_settings      import SECRET_KEY
+from django.db.models import Q
+
+from users.utils      import LoginConfirm, is_valid, JWT_ALGORITHM, \
+                             PW_REGEX, EMAIL_REGEX,  NICKNAME_REGEX
+
+class SignupView(View): 
+    def post(self,request):
+        data = json.loads(request.body)
+        
+        try:
+            assert is_valid(data['email'],EMAIL_REGEX),       "INVALID_EMAIL_FORMAT"
+            assert is_valid(data['password'],PW_REGEX),       "INVALID_PW_FORMAT"
+            assert is_valid(data['nickname'],NICKNAME_REGEX), "INVALID_NICKNAME_FORMAT"
+
+            hased_password = bcrypt.hashpw(data['password'].encode('utf-8'),bcrypt.gensalt())
+        
+            if User.objects.filter(Q(email=data['email'])|
+                                   Q(nickname=data['nickname'])
+                                   ).exists():
+            
+                return JsonResponse({"MESSAGE":"USER_ALREADY_EXISTS"},status=400)
+                
+            User.objects.create(nickname=data['nickname'], 
+                                email=data["email"],
+                                password=hased_password.decode()
+                                )   
```

Comment:
```suggestion
            User.objects.create(
                nickname=data['nickname'], 
                email=data["email"],
                password=hased_password.decode()
           )   
```
1. 이 형태가 조금 더 깔끔할 것 같네요 :)
2. `=` 기준으로 코드 정렬해주세요~

Reply:
넵 ! 수정했습니다 :+1: